File: app/src/main/java/com/yatatsu/passiveviewsample/ui/repos/ReposScreen.java
package com.yatatsu.passiveviewsample.ui.repos;


import com.yatatsu.passiveviewsample.data.model.Repository;
import com.yatatsu.passiveviewsample.ui.base.Screen;

import java.util.List;

public interface ReposScreen extends Screen {
    void showRepositories(List<Repository> repositories);
    void showError(String message);
}

File: app/src/main/java/com/yatatsu/passiveviewsample/ui/repos/ReposScreen.kt
package com.yatatsu.passiveviewsample.ui.repos


import com.yatatsu.passiveviewsample.data.model.Repository
import com.yatatsu.passiveviewsample.ui.base.Screen

interface ReposScreen : Screen {
  fun showRepositories(repositories: List<Repository>)
  fun showError(message: String)
  fun openUrl(url: String)
}
